forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestFactory.java

```diff
@@ -0,0 +1,159 @@
+            String errorString = "Registration Request JWT contained no claims";
+            log.debug("getRegistrationRequestFromJwt() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString,
+                    DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA);
+        }
+        RegistrationRequestJWTClaims registrationRequestJWTClaims =
+                new RegistrationRequestJWTClaims(registrationRequestClaimsSet,
+                        origin);
+        return registrationRequestJWTClaims;
+    }
+
+    public RegistrationRequest getRegistrationRequestFromManualRegistrationJson(String registrationRequestJson,
+                                                                                ManualRegistrationRequest manualRegistrationRequest,
+                                                                                ObjectMapper objectMapper)
+            throws DynamicClientRegistrationException {
+        try{
+            RegistrationRequest request = objectMapper.readValue(registrationRequestJson, RegistrationRequest.class);
+            request.setRedirectUris(manualRegistrationRequest.getRedirectUris());
+            String softwareStatementAssertion = manualRegistrationRequest.getSoftwareStatementAssertion();
+            if (StringUtils.isEmpty(softwareStatementAssertion)){
+                String errorMessage = "Manual Request did not contain a valid software statement";
+                log.info("getRegistrationRequestFromManualregistrationJson() {}", errorMessage);
+                throw new DynamicClientRegistrationException(errorMessage,
+                        DynamicClientRegistrationErrorType.INVALID_SOFTWARE_STATEMENT);
+            }
+            request.setSoftwareStatement(softwareStatementAssertion);
+            request.setJson(registrationRequestJson);
+
+            String ssaJwtSerialised = request.getSoftwareStatement();
+            RegistrationRequestJWTClaims softwareStatementClaims = getJwtClaimsSet(ssaJwtSerialised,
+                    JWTClaimsOrigin.SOFTWARE_STATEMENT_ASSERTION);
+            String issuer = softwareStatementClaims.getRequiredStringClaim(OpenBankingConstants.SSAClaims.ISSUER);
+            tppRegistrationService.validateSsaAgainstIssuingDirectoryJwksUri(ssaJwtSerialised, issuer);
+            DirectorySoftwareStatement regRequestSoftwareStatement =
+                    softwareStatementFactory.getSoftwareStatement(softwareStatementClaims);
+
+
+
+
+            request.setDirectorySoftwareStatement(regRequestSoftwareStatement);
+
+            return request;
+        } catch (IOException | ParseException ioe){
+            String errorMessage = "Could not map Manual Registration Request JWT fields to internal object";
+            log.info("getRegistrationRequestFromManualRegistrationJson() {}", errorMessage, ioe);
+            throw new DynamicClientRegistrationException("Could not map registration request jwt fields to " +
```

Comment:
Could be better log.error or log.warn instead of log.info? because contains the errorMessage and exception.